spark/v3.1/spark/src/main/scala/com/netease/arctic/spark/sql/catalyst/analysis/RewriteArcticCommand.scala

```diff
@@ -34,8 +32,10 @@ case class RewriteArcticCommand(sparkSession: SparkSession) extends Rule[Logical
   override def apply(plan: LogicalPlan): LogicalPlan = {
     plan match {
       // Rewrite the AlterTableDropPartition to AlterArcticTableDropPartition
-      case AlterTableDropPartition(child, parts, ifExists, purge, retainData) =>
-        AlterArcticTableDropPartition(child, parts, ifExists, purge, retainData)
+      case a@AlterTableDropPartition(r: ResolvedTable, parts, ifExists, purge, retainData) =>
+        AlterArcticTableDropPartition(a.child, parts, ifExists, purge, retainData)
+      case t@TruncateTable(r: ResolvedTable, partitionSpec) =>
+        TruncateArcticTable(t.child, partitionSpec)
       case _ => plan
```

Comment:
What if when we mixed with native iceberg table, it's also a ResolvedTable and will be rewrite as Arctic command and run failed.